### ISSUE TITLE
Quests:  Design a system for items and rooms to check task completion whenever you get the item or enter the room

### ISSUE THREAD

**Issue by JSerf02** [uchicago-cs/chiventure#1425]
We need to design a system that completes room visit and item acquisition tasks.

**Comment by JSerf02** [comment]
After discussing how this should work, we realized that the mission system includes unnecessary pointers to the objects of the missions when the string ids are really the only things necessary. Replacing these pointers with strings will help a ton with room visiting and item acquisition and it will also prevent circular dependencies with NPCs when those checks are eventually implemented. Because of this, we propose redesigning the mission module to include a single string id for the mission and an enum denoting the type of mission that it is. 

Additionally, we have narrowed down the plan for adding checks. There are functions in the player and game modules that check whether the game is in a certain room or if the player has a certain item in the inventory, so we just need to call those in `is_task_complete()`. Additionally, we need to call `is_task_complete()` in the player class function that adds items to inventory (which should be moved to game.h) and in the `move_room()` function in game.h
